### PR TITLE
Add a basic Text Fields demo to Material Gallery app.

### PR DESCRIPTION
**Pull request by mpcomplete**
BUG=https://github.com/flutter/flutter/issues/1547

### CODE REVIEW

File: examples/material_gallery/lib/demo/text_field_demo.dart [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+class TextFieldDemo extends StatefulComponent {
+  TextFieldDemo({ Key key }) : super(key: key);
+
+  TextFieldDemoState createState() => new TextFieldDemoState();
+}
+
+class TextFieldDemoState extends State<TextFieldDemo> {
+  final GlobalKey<ScaffoldState> _scaffoldKey = new GlobalKey<ScaffoldState>();
+  List<InputValue> _inputs = <InputValue>[
```
Comment (HansMuller) [inline examples/material_gallery/lib/demo/text_field_demo.dart]: This can be final

File: examples/material_gallery/lib/demo/text_field_demo.dart [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    InputValue.empty,
+    InputValue.empty,
+    InputValue.empty,
+    InputValue.empty,
+  ];
+
+  void initState() {
```
Comment (HansMuller) [inline examples/material_gallery/lib/demo/text_field_demo.dart]: Looks like this isn't needed.

File: examples/material_gallery/lib/demo/text_field_demo.dart [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+  }
+
+  void showInSnackBar(String value) {
+    _scaffoldKey.currentState.showSnackBar(new SnackBar(
+      content: new Text(value)
+    ));
+  }
+
+  void _handleInputChanged(InputValue value, int which) {
+    setState(() { _inputs[which] = value; });
```
Comment (HansMuller) [inline examples/material_gallery/lib/demo/text_field_demo.dart]: Usually we spread setState() out on three lines, even for one-liners like this.
```
setState(() {
 ...
});
```

**Comment by HansMuller** [comment]
I think this example could be a little more copy and pasteable if there was a class for collecting the users's input and if the class implementation validated its properties and generated error text to show to the user.

**Comment by HansMuller** [comment]
LGTM

**Comment by mpcomplete** [comment]
I'm working on separating the data into another class in a followup patch.

**Comment by HansMuller** [comment]
Sounds good to me.

**Pull request merged**